test/skimNtuple2017.cpp

```diff
@@ -4272,6 +4271,59 @@ int main (int argc, char** argv)
         // the HHbtag score MUST be set manually to -1
         theSmallTree.m_VBFjet2_HHbtag = -1.;
       }
+      
+      //Other jets
+      for (unsigned int iJet = 0; (iJet < theBigTree.jets_px->size ()) && (theSmallTree.m_njets < maxNjetsSaved); ++iJet)
+      {
+        // PG filter jets at will
+        if (theBigTree.PFjetID->at (iJet) < PFjetID_WP) continue; // 0 ; don't pass PF Jet ID; 1: tight, 2: tightLepVeto
+
+        // skip the H decay candiates
+        if (int (iJet) == bjet1idx ){
+          theSmallTree.m_bjet1_jecUnc = theBigTree.jets_jecUnc->at(iJet);
+          continue;
+        }else if(int (iJet) == bjet2idx){
+          theSmallTree.m_bjet2_jecUnc = theBigTree.jets_jecUnc->at(iJet);
+          continue ;
+        }
+        TLorentzVector tlv_dummyJet(
+                                    theBigTree.jets_px->at (iJet),
+                                    theBigTree.jets_py->at (iJet),
+                                    theBigTree.jets_pz->at (iJet),
+                                    theBigTree.jets_e->at (iJet)
+                                   );
+
+        // Apply PUjetID only to jets with pt < 50 GeV ( https://twiki.cern.ch/twiki/bin/view/CMS/HiggsToTauTauWorkingLegacyRun2#Jets )
+        // PU jet ID WP = 2: loose
+        if (PUjetID_WP > -1)
+        {
+          if ( !(CheckBit(theBigTree.jets_PUJetIDupdated_WP->at(iJet), PUjetID_WP)) && tlv_dummyJet.Pt()<50.) continue;
+        }
+
+        // Apply further cleaning for 2017 noisy jets, as suggested by HTT group: https://twiki.cern.ch/twiki/bin/view/CMS/HiggsToTauTauWorkingLegacyRun2#Jets
+        // The noisy jets to be removed are defined as: 20 < pt < 50 && abs(eta) > 2.65 && abs(eta) < 3.139
+        //if ( tlv_dummyJet.Pt()<50. && fabs(tlv_dummyJet.Eta())>2.65 && fabs(tlv_dummyJet.Eta()<3.139) ) continue; //Commented during March20 sync
+
+        // remove jets that overlap with the tau selected in the leg 1 and 2
+        if (tlv_firstLepton.DeltaR(tlv_dummyJet) < lepCleaningCone){
+          theSmallTree.m_dau1_jecUnc = theBigTree.jets_jecUnc->at(iJet);
```

Comment:
Please, remove this line with the uncertainty, it's already taken care.
Here you just want to skip the jet if it overlaps with tau1 or tau2.